DuckDuckGo/AutocompleteViewController.swift

```diff
@@ -53,7 +53,11 @@ class AutocompleteViewController: UIViewController {
             view.backgroundColor = showBackground ? UIColor.black.withAlphaComponent(0.2) : UIColor.clear
         }
     }
-    
+
+    var selectedSuggestion: Suggestion? {
+        (selectedItem >= 0 && selectedItem < suggestions.count) ? suggestions[selectedItem] : nil
```

Comment:
One neat trick I learned recently was to use `indices` to do bounds checking, like `suggestions.indices.contains(selectedItem)`. IMO it's easier to read than comparison statements.

I've been thinking about adding an extension to `Collection` to do safe lookups, like:

```
subscript (safe index: Index) -> Element? {
    return indices.contains(index) ? self[index] : nil
}
```

None of this is a blocker, just thinking out loud. 🙂